### ISSUE TITLE
OIDC login: add "auto redirect" option for SSO login

### ISSUE THREAD

**Issue by iWangJiaxiang** [goharbor/harbor#11077]
**Is your feature request related to a problem? Please describe.**

I am using SSO (single sign on) for all services in my team. The harbor login page should be automatically redirected to SSO login page for authentication. However, the harbor login page just offers an OIDC button to manually redirect to SSO login page.

**Describe the solution you'd like**

Add a function for automatic redirection from harbor login page to OIDC login page by:

1. Add an option `auto redirect` on OIDC setting page to trigger  automatic/manual redirection to SSO login page.
2. Add a http request parameter `autoredirect=false` for `/harbor/sign-in` harbor login page. So if automatic redirection is set, user/admin will have the choice to choose direct login. 
3. Document the `autoredirect` request parameter in OIDC setting page to tell the admin using this parameter manully to direct login.

**Describe the main design/architecture of your solution**

1. A boolean (default: false) `auto_redirect` property added in OIDC setting model.
2. A boolean (default: true)  `autoredirect` parameter processing in harbor login server controller. And add some codes like this:

```Csharp
if OIDC_model.auto_redirect==true && autoredirect==true
{
    // redirect to OIDC login page
}
```

**Describe the development plan you've considered**

I wish I could implementation this feature but I am not familiar with Golang.

**Additional context**

SSO is very useful for enterprise. This feature would be an improvement for enterprise SSO experience.

Thank you!

**Comment by reasonerjt** [comment]
I think there may be some direction case we need to clarify,

for example if for some reason user's session expires he will be redirected to login page of OIDC provider after he inputs the credentials and move with the token exchange flow, it may be hard to redirect to the page where he was visiting.

**Comment by iWangJiaxiang** [comment]
@reasonerjt Yes. Right now some of SSO services in my team just **redirect back to home page** of service.

I tested your case using the current OIDC workflow (manual redirection) as well as direct login workflow in harbor. And I fould the harbor would redirect back to `/harbor/projects` page anyway. No matter whether OIDC was used.

### Test Procedure:

1. Copy a valid url path `https://harbor.example.com/harbor/projects/1/summary`.
2. Open a new browser that needs authentication.
3. Visit that url.
4. Login with direct login workflow or OIDC workflow.
5. The browser will redirect to `https://harbor.example.com/harbor/projects` anyway.

### Summary

I think the correct redirection issue is not relative to my proposal.

**Comment by xaleeks** [comment]
Let's discuss this for v2.2 @reasonerjt

**Comment by reasonerjt** [comment]
@xaleeks I don't think this is covered in v2.2 plan,  this requirement mainly focuses on UX improvement, it will not block any workflow if it is not implemented, hence I'm putting it in backlog
